spec.html

```diff
@@ -20805,8 +20798,8 @@ <h1>Runtime Semantics: ClassDefinitionEvaluation</h1>
         1. Set the running execution context's LexicalEnvironment to _classScope_.
         1. Let _constructorInfo_ be ! DefineMethod of _constructor_ with arguments _proto_ and _constructorParent_.
         1. Let _F_ be _constructorInfo_.[[Closure]].
-        1. If |ClassHeritage_opt| is present, set _F_.[[ConstructorKind]] to ~derived~.
         1. Perform MakeConstructor(_F_, *false*, _proto_).
+        1. If |ClassHeritage_opt| is present, set _F_.[[ConstructorKind]] to ~derived~.
```

Comment:
Was this line moved because it feels odd to set [[ConstructorKind]] on something that doesn't yet have a [[Construct]]?

Reply:
`MakeConstructor` has `Set _F_.[[ConstructorKind]] to ~base~`, so this line has to come after the invocation of `MakeConstructor` to (sometimes) override that setting.